sources/Core/DSL/Constant`2.cs

```diff
@@ -0,0 +1,51 @@
+using System.Runtime.CompilerServices;
+
+namespace Silk.NET.Core;
+
+/// <summary>
+/// Represents a constant that optionally can be represented as an enum. This allows both constant integers and
+/// enumerated members to be used by the same parameter without a cast.
+/// </summary>
+/// <param name="Value">The underlying value.</param>
+/// <typeparam name="TUnderlying">The underlying value type.</typeparam>
+/// <typeparam name="TEnum1">
+/// The enum that can be used in place of a <typeparamref name="TUnderlying"/> constant.
+/// </typeparam>
+public readonly record struct Constant<TUnderlying, TEnum1>(TUnderlying Value)
```

Comment:
Need thoughts on what the name of this should be, as in theory we might want to use this for OpenGL handles as well e.g. `Handle<uint, Program>`. Maybe we could all this `Alt`?